### PR TITLE
feat: pangolin faucet

### PR DESCRIPTION
**Pull request by JayJay1024**
and refactor crab faucet

### CODE REVIEW

File: airdrop/api/grant.ts [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
+        },
+        pangolin: {
+          callback: "/api/pangolin/authorization",
+        },
+      },
+    },
+  };
   config.github.key = process.env.GITHUB_OAUTH_APP_KEY;
-  config.github.secret = process.env.GITHUB_OAUTH_APP_SECRET;
+  config.github.secret = process.env.GITHUB_OAUTH_APP_KEY;
```
Comment (fewensa) [inline airdrop/api/grant.ts:26]: I think there used the wrong variable. should be using `process.env.GITHUB_OAUTH_APP_SECRET`

File: airdrop/api/pangolin/airdrop/state.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      message: "Connection Redis failed",
+      data: {
+        state: "",
+      },
+    };
+    res.end(JSON.stringify(body, null, 2));
+    return;
+  }
+
+  // const chainName = data.chain.toUpperCase().trim();
```
Comment (fewensa) [inline airdrop/api/pangolin/airdrop/state.ts]: Clean these comments?

Comment (cinlk) [reply airdrop/api/pangolin/airdrop/state.ts]: > Clean these comments?

done

File: airdrop/api/grant.ts (hunk 1 above)
Comment (cinlk) [reply airdrop/api/grant.ts]: done

**Review by fewensa — APPROVED** [review]

**Review by 1022ranji — APPROVED** [review]

**Pull request merged**